### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by jesusbv**
This is a first step into improving the unit tests, I plan to add more specific/meaningful tests 

This adds 99% coverage, as the SMT ip issue fails, the test is ready but commented 
once uncommented, coverage should be 100%, meaning these tests would have caught
that issue 

```bash
Name                                 Stmts   Miss  Cover   Missing
------------------------------------------------------------------
lib/cloudregister/__init__.py            0      0   100%
lib/cloudregister/amazonec2.py          34      0   100%
lib/cloudregister/googlece.py           23      0   100%
lib/cloudregister/msftazure.py          67      0   100%
lib/cloudregister/registerutils.py     946      0   100%
lib/cloudregister/smt.py               143      4    97%   256-263
------------------------------------------------------------------
TOTAL                                 1213      4    99%

=================================== 210 passed, 5 warnings in 55.35s ====================================
```

### CODE REVIEW

**Comment by jesusbv** [comment]
I will add also the docstrings for the test methods

**Review by jgleissner — APPROVED** [review]
LGTM

**Review by smarlowucf — APPROVED** [review]
Yay tests 🎉

File: tests/test_registerutils.py [hunk 1]
```diff
@@ -458,8 +487,2449 @@ def test_clean_host_file_no_empty_bottom_lines_smt_entry_is_last():
     assert m().write.mock_calls == expected_write_calls
 
 
+def test_clean_host_file_raised_exception():
+    hosts_content = ""
+    expected_cleaned_hosts = """
+# simulates hosts file containing the ipv6 we are looking for in the test
```
Comment (rjschwei) [inline tests/test_registerutils.py:493]: this says "ipv6" but what follows are ipv4 address, meaning this duplicates the mistake that is already in the tests.

**Review by rjschwei — CHANGES_REQUESTED** [review]
Didn't get to review all the new tests, ran out of time/steam

File: lib/cloudregister/registerutils.py [hunk 2]
```diff
@@ -657,8 +653,8 @@ def get_installed_products():
             errMsg = 'zypper product query returned with zypper code %d'
             logging.error(errMsg % cmd.returncode)
             return products
-    except OSError:
-        errMsg = 'Could not get product list %s' % cmd[1]
+    except OSError as err:
+        errMsg = 'Could not get product list: %s' % err
```
[the remark this answers is not in the document]
Comment (jesusbv) [reply lib/cloudregister/registerutils.py:657]: Understood, as it was that raised an exception if `subprocess.Popen` produced `OSError`
as cmd was not defined

Will update it to log the command

File: tests/test_smt.py [hunk 3]
```diff
@@ -152,6 +153,60 @@ def test_get_cert_no_match_cert(mock_cert_pull, mock_logging, mock_load_cert):
...
+
+# ----------------------------------------------------------------------------
+@patch.object(SMT, 'get_cert')
+def test_write_cert(mock_get_cert):
+    """Received cert with different fingerprint"""
+    with open('tests/data/cert.pem', 'r') as cert_file:
+        mock_get_cert.return_value = cert_file.read()
+
+    smt = SMT(etree.fromstring(smt_data_ipv46))
+    assert smt.write_cert('/tmp') == '/tmp/registration_server_fc00__1.pem'
```
[the remark this answers is not in the document]
Comment (jesusbv) [reply tests/test_smt.py:206]: Yes, will update
I thought I had changed all the writing to disk tests,
will check I did not miss more of those

**Review by rjschwei — APPROVED** [review]

**Pull request merged**